be/src/storage/vectorized/compaction.cpp

```diff
@@ -188,49 +189,66 @@ Status Compaction::merge_rowsets(int64_t mem_limit, Statistics* stats_output) {
         }
 #endif
 
-        chunk->reset();
-        Status status = reader.get_next(chunk.get());
-        if (!status.ok()) {
-            if (status.is_end_of_file()) {
-                break;
-            } else {
-                return Status::InternalError("reader get_next error.");
+        bool bg_worker_stopped = ExecEnv::GetInstance()->storage_engine()->bg_worker_stopped();
+        while (!bg_worker_stopped) {
```

Comment:
```suggestion
        while (!ExecEnv::GetInstance()->storage_engine()->bg_worker_stopped()) {
```
And remove the variable  `bg_worker_stopped`